From 065fdcb0b0a80742445abd3255bbf2b368a601fc Mon Sep 17 00:00:00 2001
From: yanxurui <yanxurui@cloudin.com>
Date: Tue, 21 Mar 2017 14:36:08 +0800
Subject: [PATCH] hack: allow session-id to contain dot

---
 ngx_http_upload_module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ngx_http_upload_module.c b/ngx_http_upload_module.c
index e7be6f5..0448f0e 100644
--- a/ngx_http_upload_module.c
+++ b/ngx_http_upload_module.c
@@ -3904,7 +3904,7 @@ ngx_http_upload_validate_session_id(ngx_str_t *session_id) {
 
     while(p != q) {
         if(!((*p >= '0' && *p <= '9') || (*p >= 'a' && *p <= 'z') || (*p >= 'A' && *p <= 'Z')
-            || *p == '_' || *p == '-'))
+            || *p == '_' || *p == '-' || *p == '.'))
         {
             return NGX_ERROR;
         }
--
2.11.0

